File: All Topics/1-50/1-TwoSum.ts
function twoSum(nums: number[], target: number): number[] {
  const numsPossibles = [...nums];
  for (let i = 0; i < nums.length; i++) {
    const actualNumber = numsPossibles[0];
    numsPossibles.shift();
    if (numsPossibles.indexOf(target - actualNumber) !== -1) {
      return [i, nums.lastIndexOf(target - actualNumber)];
    }
  }

  return [1, 2];
}

File: All-Topics/1-50/1-TwoSum.cpp



class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map<int,int> map;


        for(int i=0;i<nums.size();i++){

            int find=target-nums[i];

            if(map.count(find)){
                return {i,map[find]};
            }

            map[nums[i]]=i;


        }

        return {};
    }
};